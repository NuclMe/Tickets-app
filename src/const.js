export const TRANSFER_OPTIONS = [
  { label: 'Все', value: 'all' },
  { label: 'Без пересадок', value: 0 },
  { label: '1 пересадка', value: 1 },
  { label: '2 пересадки', value: 2 },
  { label: '3 пересадки', value: 3 },
];

const size = {
  mobileS: '320px',
  mobileM: '375px',
  mobileL: '425px',
  tablet: '768px',
  laptop: '1024px',
  laptopL: '1440px',
  desktop: '2560px',
};
export const device = {
  mobileS: `(min-width: ${size.mobileS})`,
  mobileM: `(min-width: ${size.mobileM})`,
  mobileL: `(min-width: ${size.mobileL})`,
  tablet: `(max-width: ${size.tablet})`,
  laptop: `(min-width: ${size.laptop})`,
  laptopL: `(min-width: ${size.laptopL})`,
  desktop: `(min-width: ${size.desktop})`,
  desktopL: `(min-width: ${size.desktop})`,
};
