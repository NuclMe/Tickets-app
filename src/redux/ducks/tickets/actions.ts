import { types } from './types';

export const fetchTickets = () => ({ type: types.FETCH_TICKETS });
export const fetchTicketsSuccess = (tickets) => ({
  type: types.FETCH_TICKETS_SUCCESS,
  payload: tickets,
});
export const fetchTicketsError = (error) => ({
  type: types.FETCH_TICKETS_ERROR,
  error,
});

export const setTransferFilter = (filter) => ({
  type: types.SET_TRANSFER_FILTER,
  payload: filter,
});
