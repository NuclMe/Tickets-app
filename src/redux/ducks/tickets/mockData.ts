import type { ITicket } from '../../../types';

export const mockTickets: ITicket[] = [
  {
    price: 35230,
    carrier: 'SU',
    segments: [
      {
        origin: 'MOW',
        destination: 'HKT',
        date: '2018-05-18T02:00:00.000Z',
        stops: [],
        duration: 400,
      },
      {
        origin: 'HKT',
        destination: 'MOW',
        date: '2018-05-18T08:30:00.000Z',
        stops: [],
        duration: 300,
      },
    ],
  },
  {
    price: 30000,
    carrier: 'S7',
    segments: [
      {
        origin: 'MOW',
        destination: 'HKT',
        date: '2018-05-18T02:00:00.000Z',
        stops: ['LED'],
        duration: 600,
      },
      {
        origin: 'HKT',
        destination: 'MOW',
        date: '2018-05-18T08:30:00.000Z',
        stops: ['LED'],
        duration: 200,
      },
    ],
  },
  {
    price: 25000,
    carrier: 'TK',
    segments: [
      {
        origin: 'MOW',
        destination: 'HKT',
        date: '2018-05-18T02:00:00.000Z',
        stops: ['IST', 'DXB'],
        duration: 750,
      },
      {
        origin: 'HKT',
        destination: 'MOW',
        date: '2018-05-18T08:30:00.000Z',
        stops: ['IST', 'DXB'],
        duration: 650,
      },
    ],
  },
  {
    price: 40000,
    carrier: 'BA',
    segments: [
      {
        origin: 'MOW',
        destination: 'HKT',
        date: '2018-05-18T02:00:00.000Z',
        stops: ['LED'],
        duration: 500,
      },
      {
        origin: 'HKT',
        destination: 'MOW',
        date: '2018-05-18T08:30:00.000Z',
        stops: ['LED', 'AMS'],
        duration: 450,
      },
    ],
  },
  {
    price: 45000,
    carrier: 'LH',
    segments: [
      {
        origin: 'MOW',
        destination: 'HKT',
        date: '2018-05-18T02:00:00.000Z',
        stops: ['FRA', 'BKK', 'SIN'],
        duration: 800,
      },
      {
        origin: 'HKT',
        destination: 'MOW',
        date: '2018-05-18T08:30:00.000Z',
        stops: ['SIN', 'BKK', 'FRA'],
        duration: 750,
      },
    ],
  },
];
