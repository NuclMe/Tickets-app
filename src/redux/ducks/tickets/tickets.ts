import { createSelector } from 'reselect';
import type { RootState } from '../../root-reducer';
import { types } from './types';
import type { ITicket } from '../../../types';
import { call, put, takeLatest } from 'redux-saga/effects';
import axios from 'axios';

const initialState = {
  tickets: [],
  loading: false,
  error: null,
  selectedTransfer: 'all',
};

const ticketsSelector = (state: RootState) => state.tickets.tickets;
const transferSelector = (state: RootState) => state.tickets.selectedTransfer;

export const selectFilteredTickets = createSelector(
  [ticketsSelector, transferSelector],
  (tickets: ITicket[], selectedTransfer: string) => {
    if (selectedTransfer === 'all') return tickets;
    return tickets.filter((ticket: ITicket) =>
      ticket.segments.every(
        (seg) => seg.stops.length === Number(selectedTransfer)
      )
    );
  }
);

export const fetchTickets = () => ({ type: types.FETCH_TICKETS });
export const fetchTicketsSuccess = (tickets: ITicket[]) => ({
  type: types.FETCH_TICKETS_SUCCESS,
  payload: tickets,
});
export const fetchTicketsError = (error: string) => ({
  type: types.FETCH_TICKETS_ERROR,
  error,
});
export const setTransferFilter = (filter: number | string) => ({
  type: types.SET_TRANSFER_FILTER,
  payload: filter,
});

export const reducer = (state = initialState, action: any) => {
  switch (action.type) {
    case types.FETCH_TICKETS:
      return { ...state, loading: true, error: null };
    case types.FETCH_TICKETS_SUCCESS:
      return { ...state, loading: false, tickets: action.payload };
    case types.FETCH_TICKETS_ERROR:
      return { ...state, loading: false, error: action.error };
    case types.SET_TRANSFER_FILTER:
      return { ...state, selectedTransfer: action.payload };
    default:
      return state;
  }
};
function* fetchTicketsSaga(): Generator {
  try {
    let allTickets: ITicket[] = [];
    let stop = false;
    let retries = 0;
    const maxRetries = 10;

    while (!stop && retries < maxRetries) {
      try {
        const ticketsRes = yield call(axios.get, '/mock-api/tickets.json');
        allTickets = allTickets.concat(ticketsRes.data.tickets);
        stop = ticketsRes.data.stop;
      } catch (err) {
        console.error(err);
        retries += 1;
        yield call(() => new Promise((res) => setTimeout(res, 500)));
        continue;
      }
    }

    yield put(fetchTicketsSuccess(allTickets));
  } catch (error: any) {
    yield put(fetchTicketsError(error.message));
  }
}

export function* ticketSaga() {
  yield takeLatest(types.FETCH_TICKETS, fetchTicketsSaga);
}
