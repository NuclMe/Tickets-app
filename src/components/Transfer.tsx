import React from 'react';
import Box from '@mui/material/Box';
import Typography from '@mui/material/Typography';
import FormGroup from '@mui/material/FormGroup';
import FormControlLabel from '@mui/material/FormControlLabel';
import Checkbox from '@mui/material/Checkbox';
import { TRANSFER_OPTIONS } from '../const';

export const Transfer = () => {
  return (
    <Box
      sx={{
        display: 'flex',
        flexDirection: 'column',
        alignItems: 'flex-start',
        border: '1px solid lightgray',
        borderRadius: '8px',
        boxShadow: '0 2px 4px rgba(0, 0, 0, 0.8)',
        backgroundColor: 'white',
        flex: '1',
        padding: '20px 0',
      }}
    >
      <Typography
        variant="h6"
        sx={{
          marginLeft: '20px',
          marginBottom: '10px',
          fontSize: '16px',
          '@media (max-width: 768px)': {
            fontSize: '14px',
          },
        }}
      >
        КОЛИЧЕСТВО ПЕРЕСАДОК
      </Typography>
      <FormGroup
        sx={{
          width: '100%',
          overflow: 'hidden',
          '@media (max-width: 768px)': {
            fontSize: '10px',
          },
        }}
      >
        {TRANSFER_OPTIONS.map((option) => (
          <FormControlLabel
            key={option.value}
            control={
              <Checkbox
                sx={{
                  '& .MuiSvgIcon-root': { fontSize: 28 },
                  p: 0,
                  marginRight: '10px',
                }}
              />
            }
            label={option.label}
            sx={{
              width: '100%',
              paddingLeft: '20px',
              paddingBottom: '10px',
              paddingTop: '10px',
              marginRight: '0px',
              marginLeft: '0px',
              '&:hover': {
                backgroundColor: 'rgba(173, 216, 230, 0.4)',
              },
              '@media (max-width: 768px)': {
                '& .MuiTypography-root': {
                  fontSize: '14px',
                },
              },
            }}
          />
        ))}
      </FormGroup>
    </Box>
  );
};
