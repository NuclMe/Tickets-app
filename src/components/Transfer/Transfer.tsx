import Box from '@mui/material/Box';
import FormControlLabel from '@mui/material/FormControlLabel';
import FormGroup from '@mui/material/FormGroup';
import Radio from '@mui/material/Radio';
import Typography from '@mui/material/Typography';
import { Formik, Form, Field } from 'formik';
import type { FC } from 'react';
import { useDispatch, useSelector } from 'react-redux';

import { TRANSFER_OPTIONS } from '../../const';
import { setTransferFilter } from '../../redux/ducks/tickets/tickets';
import type { RootState } from '../../redux/root-reducer';
import { device, fontSizes } from '../../styledVars';

export const Transfer: FC = () => {
  const dispatch = useDispatch();
  const selectedTransfer = useSelector(
    (state: RootState) => state.tickets.selectedTransfer
  );
  return (
    <Box
      sx={{
        display: 'flex',
        flexDirection: 'column',
        alignItems: 'flex-start',
        border: '1px solid lightgray',
        borderRadius: '8px',
        boxShadow: '0 2px 4px rgba(0, 0, 0, 0.8)',
        backgroundColor: 'white',
        flex: '1',
        padding: '20px 0',
        maxHeight: '340px',
      }}
    >
      <Typography
        variant="h6"
        sx={{
          marginLeft: '20px',
          marginBottom: '10px',
          fontSize: fontSizes.base,
          [`${device.tablet}`]: {
            fontSize: fontSizes.xs,
          },
        }}
      >
        КОЛИЧЕСТВО ПЕРЕСАДОК
      </Typography>
      <Formik
        initialValues={{ transfer: selectedTransfer || 'all' }}
        enableReinitialize
        onSubmit={(values) => {
          dispatch(setTransferFilter(values.transfer));
        }}
      >
        {({ values, setFieldValue }) => (
          <Form style={{ width: '100%', overflow: 'hidden' }}>
            <FormGroup>
              {TRANSFER_OPTIONS.map((option) => (
                <FormControlLabel
                  key={option.value}
                  control={
                    <Field
                      as={Radio}
                      type="radio"
                      name="transfer"
                      value={option.value}
                      checked={String(values.transfer) === String(option.value)}
                      onChange={() => {
                        setFieldValue('transfer', option.value);
                        dispatch(setTransferFilter(option.value));
                      }}
                    />
                  }
                  label={option.label}
                  sx={{
                    width: '100%',
                    paddingLeft: '20px',
                    paddingBottom: '10px',
                    paddingTop: '10px',
                    marginRight: '0px',
                    marginLeft: '0px',
                    '&:hover': {
                      backgroundColor: 'rgba(173, 216, 230, 0.4)',
                    },
                    [`${device.tablet}`]: {
                      '& .MuiTypography-root': {
                        fontSize: fontSizes.xs,
                      },
                    },
                  }}
                />
              ))}
            </FormGroup>
          </Form>
        )}
      </Formik>
    </Box>
  );
};
