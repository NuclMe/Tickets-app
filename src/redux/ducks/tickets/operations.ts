import { call, put, takeLatest } from 'redux-saga/effects';
import axios from 'axios';
import { types } from './types';
import { fetchTicketsSuccess, fetchTicketsError } from './actions';

function* fetchTicketsSaga() {
  try {
    let allTickets = [];
    let stop = false;
    let retries = 0;
    const maxRetries = 10;

    while (!stop && retries < maxRetries) {
      try {
        const ticketsRes = yield call(axios.get, '/mock-api/tickets.json');
        allTickets = allTickets.concat(ticketsRes.data.tickets);
        stop = ticketsRes.data.stop;
      } catch (err) {
        retries += 1;
        yield call(() => new Promise((res) => setTimeout(res, 500)));
        continue;
      }
    }

    yield put(fetchTicketsSuccess(allTickets));
  } catch (error) {
    yield put(fetchTicketsError(error.message));
  }
}

export function* ticketSaga() {
  yield takeLatest(types.FETCH_TICKETS, fetchTicketsSaga);
}
