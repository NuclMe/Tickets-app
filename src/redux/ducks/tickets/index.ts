import { fetchTickets } from './actions';
import { reducer } from './reducer';
import { ticketSaga } from './operations';
import { selectFilteredTickets } from './selectors';

export { fetchTickets, reducer, ticketSaga, selectFilteredTickets };
