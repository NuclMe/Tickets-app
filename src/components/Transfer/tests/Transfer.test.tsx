import { configureStore } from '@reduxjs/toolkit';
import { render, screen, fireEvent } from '@testing-library/react';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import { describe, expect, it, vi } from 'vitest';

import { TRANSFER_OPTIONS } from '../../../const';
import { reducer as ticketsReducer } from '../../../redux/ducks/tickets/tickets';
import { Transfer } from '../Transfer';

const createTestStore = (preloadedState = {}) => {
  const store = configureStore({
    reducer: {
      tickets: ticketsReducer,
    },
    preloadedState: {
      tickets: {
        tickets: [],
        loading: false,
        error: null,
        selectedTransfer: 'all',
        ...preloadedState,
      },
    },
  });

  store.dispatch = vi.fn(store.dispatch);
  return store;
};

const TestWrapper = ({ children, initialState = {} }) => {
  const store = createTestStore(initialState);
  return (
    <Provider store={store}>
      <BrowserRouter>{children}</BrowserRouter>
    </Provider>
  );
};

describe('Transfer', () => {
  it('renders the Transfer component via snapshot', () => {
    const { container } = render(
      <TestWrapper>
        <Transfer />
      </TestWrapper>
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render all transfer options from TRANSFER_OPTIONS', () => {
    render(
      <TestWrapper>
        <Transfer />
      </TestWrapper>
    );

    TRANSFER_OPTIONS.forEach((option) => {
      expect(screen.getByLabelText(option.label)).toBeInTheDocument();
    });
  });

  it('should initialize with selectedTransfer from Redux state', () => {
    render(
      <TestWrapper initialState={{ selectedTransfer: '1' }}>
        <Transfer />
      </TestWrapper>
    );

    const radioButton = screen.getByDisplayValue('1');
    expect(radioButton).toBeChecked();
  });

  it('should handle radio button change and dispatch action (lines 69-71)', () => {
    const store = createTestStore();
    const dispatchSpy = vi.spyOn(store, 'dispatch');

    render(
      <Provider store={store}>
        <BrowserRouter>
          <Transfer />
        </BrowserRouter>
      </Provider>
    );

    const oneStopRadio = screen.getByDisplayValue('1');

    fireEvent.click(oneStopRadio);

    expect(dispatchSpy).toHaveBeenCalledWith({
      type: 'SET_TRANSFER_FILTER',
      payload: 1,
    });
  });

  it('should update field value when radio button is clicked (line 70)', () => {
    render(
      <TestWrapper>
        <Transfer />
      </TestWrapper>
    );

    expect(screen.getByDisplayValue('all')).toBeChecked();

    const zeroStopsRadio = screen.getByDisplayValue('0');
    fireEvent.click(zeroStopsRadio);

    expect(zeroStopsRadio).toBeChecked();
    expect(screen.getByDisplayValue('all')).not.toBeChecked();
  });

  it('should dispatch setTransferFilter with correct value (line 71)', () => {
    const store = createTestStore();
    const dispatchSpy = vi.spyOn(store, 'dispatch');

    render(
      <Provider store={store}>
        <BrowserRouter>
          <Transfer />
        </BrowserRouter>
      </Provider>
    );

    const twoStopsRadio = screen.getByDisplayValue('2');
    fireEvent.click(twoStopsRadio);

    expect(dispatchSpy).toHaveBeenCalledWith({
      type: 'SET_TRANSFER_FILTER',
      payload: 2,
    });
  });

  it('should maintain form state consistency with Redux state', () => {
    const store = createTestStore({ selectedTransfer: '0' });
    const dispatchSpy = vi.spyOn(store, 'dispatch');

    render(
      <Provider store={store}>
        <BrowserRouter>
          <Transfer />
        </BrowserRouter>
      </Provider>
    );

    expect(screen.getByDisplayValue('0')).toBeChecked();

    const oneStopRadio = screen.getByDisplayValue('1');
    fireEvent.click(oneStopRadio);

    expect(oneStopRadio).toBeChecked();
    expect(dispatchSpy).toHaveBeenCalledWith({
      type: 'SET_TRANSFER_FILTER',
      payload: 1,
    });
  });

  it('should handle enableReinitialize correctly', () => {
    const { rerender } = render(
      <TestWrapper initialState={{ selectedTransfer: 'all' }}>
        <Transfer />
      </TestWrapper>
    );

    expect(screen.getByDisplayValue('all')).toBeChecked();

    rerender(
      <TestWrapper initialState={{ selectedTransfer: '2' }}>
        <Transfer />
      </TestWrapper>
    );

    expect(screen.getByDisplayValue('2')).toBeChecked();
  });
});
