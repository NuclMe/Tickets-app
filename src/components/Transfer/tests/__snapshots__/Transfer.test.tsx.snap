// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Transfer > renders the Transfer component via snapshot 1`] = `
<div
  class="MuiBox-root css-1ox7709"
>
  <h6
    class="MuiTypography-root MuiTypography-h6 css-1cn46yb-MuiTypography-root"
  >
    КОЛИЧЕСТВО ПЕРЕСАДОК
  </h6>
  <form
    action="#"
    style="width: 100%; overflow: hidden;"
  >
    <div
      class="MuiFormGroup-root css-1mac7pt-MuiFormGroup-root"
    >
      <label
        class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-1az72d7-MuiFormControlLabel-root"
      >
        <span
          class="MuiButtonBase-root MuiRadio-root MuiRadio-colorPrimary PrivateSwitchBase-root MuiRadio-root MuiRadio-colorPrimary Mui-checked MuiRadio-root MuiRadio-colorPrimary css-1q72ply-MuiButtonBase-root-MuiSwitchBase-root-MuiRadio-root"
        >
          <input
            checked=""
            class="PrivateSwitchBase-input css-12xagqm-MuiSwitchBase-root"
            name="transfer"
            type="radio"
            value="all"
          />
          <span
            class="css-9oxshb-MuiRadioButtonIcon-root"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-iy8fvg-MuiSvgIcon-root-MuiRadioButtonIcon-root"
              data-testid="RadioButtonUncheckedIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
              />
            </svg>
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1pqw2bl-MuiSvgIcon-root-MuiRadioButtonIcon-root"
              data-testid="RadioButtonCheckedIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
              />
            </svg>
          </span>
        </span>
        <span
          class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-rizt0-MuiTypography-root"
        >
          Все
        </span>
      </label>
      <label
        class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-1az72d7-MuiFormControlLabel-root"
      >
        <span
          class="MuiButtonBase-root MuiRadio-root MuiRadio-colorPrimary PrivateSwitchBase-root MuiRadio-root MuiRadio-colorPrimary MuiRadio-root MuiRadio-colorPrimary css-1q72ply-MuiButtonBase-root-MuiSwitchBase-root-MuiRadio-root"
        >
          <input
            class="PrivateSwitchBase-input css-12xagqm-MuiSwitchBase-root"
            name="transfer"
            type="radio"
            value="0"
          />
          <span
            class="css-9oxshb-MuiRadioButtonIcon-root"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-iy8fvg-MuiSvgIcon-root-MuiRadioButtonIcon-root"
              data-testid="RadioButtonUncheckedIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
              />
            </svg>
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-cpa8b4-MuiSvgIcon-root-MuiRadioButtonIcon-root"
              data-testid="RadioButtonCheckedIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
              />
            </svg>
          </span>
        </span>
        <span
          class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-rizt0-MuiTypography-root"
        >
          Без пересадок
        </span>
      </label>
      <label
        class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-1az72d7-MuiFormControlLabel-root"
      >
        <span
          class="MuiButtonBase-root MuiRadio-root MuiRadio-colorPrimary PrivateSwitchBase-root MuiRadio-root MuiRadio-colorPrimary MuiRadio-root MuiRadio-colorPrimary css-1q72ply-MuiButtonBase-root-MuiSwitchBase-root-MuiRadio-root"
        >
          <input
            class="PrivateSwitchBase-input css-12xagqm-MuiSwitchBase-root"
            name="transfer"
            type="radio"
            value="1"
          />
          <span
            class="css-9oxshb-MuiRadioButtonIcon-root"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-iy8fvg-MuiSvgIcon-root-MuiRadioButtonIcon-root"
              data-testid="RadioButtonUncheckedIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
              />
            </svg>
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-cpa8b4-MuiSvgIcon-root-MuiRadioButtonIcon-root"
              data-testid="RadioButtonCheckedIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
              />
            </svg>
          </span>
        </span>
        <span
          class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-rizt0-MuiTypography-root"
        >
          1 пересадка
        </span>
      </label>
      <label
        class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-1az72d7-MuiFormControlLabel-root"
      >
        <span
          class="MuiButtonBase-root MuiRadio-root MuiRadio-colorPrimary PrivateSwitchBase-root MuiRadio-root MuiRadio-colorPrimary MuiRadio-root MuiRadio-colorPrimary css-1q72ply-MuiButtonBase-root-MuiSwitchBase-root-MuiRadio-root"
        >
          <input
            class="PrivateSwitchBase-input css-12xagqm-MuiSwitchBase-root"
            name="transfer"
            type="radio"
            value="2"
          />
          <span
            class="css-9oxshb-MuiRadioButtonIcon-root"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-iy8fvg-MuiSvgIcon-root-MuiRadioButtonIcon-root"
              data-testid="RadioButtonUncheckedIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
              />
            </svg>
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-cpa8b4-MuiSvgIcon-root-MuiRadioButtonIcon-root"
              data-testid="RadioButtonCheckedIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
              />
            </svg>
          </span>
        </span>
        <span
          class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-rizt0-MuiTypography-root"
        >
          2 пересадки
        </span>
      </label>
      <label
        class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-1az72d7-MuiFormControlLabel-root"
      >
        <span
          class="MuiButtonBase-root MuiRadio-root MuiRadio-colorPrimary PrivateSwitchBase-root MuiRadio-root MuiRadio-colorPrimary MuiRadio-root MuiRadio-colorPrimary css-1q72ply-MuiButtonBase-root-MuiSwitchBase-root-MuiRadio-root"
        >
          <input
            class="PrivateSwitchBase-input css-12xagqm-MuiSwitchBase-root"
            name="transfer"
            type="radio"
            value="3"
          />
          <span
            class="css-9oxshb-MuiRadioButtonIcon-root"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-iy8fvg-MuiSvgIcon-root-MuiRadioButtonIcon-root"
              data-testid="RadioButtonUncheckedIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
              />
            </svg>
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-cpa8b4-MuiSvgIcon-root-MuiRadioButtonIcon-root"
              data-testid="RadioButtonCheckedIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
              />
            </svg>
          </span>
        </span>
        <span
          class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-rizt0-MuiTypography-root"
        >
          3 пересадки
        </span>
      </label>
    </div>
  </form>
</div>
`;
